misc-utils/lsfd-sock-xinfo.c

```diff
@@ -52,41 +55,111 @@ static void load_xinfo_from_proc_udp6(ino_t netns_inode);
 static void load_xinfo_from_proc_udplite6(ino_t netns_inode);
 static void load_xinfo_from_proc_raw6(ino_t netns_inode);
 static void load_xinfo_from_proc_netlink(ino_t netns_inode);
+static void load_xinfo_from_proc_packet(ino_t netns_inode);
 
 static int self_netns_fd = -1;
 static struct stat self_netns_sb;
 
 static void *xinfo_tree;	/* for tsearch/tfind */
 static void *netns_tree;
 
+struct iface {
+	unsigned int index;
+	char name[IF_NAMESIZE];
+};
+
+static const char *get_iface_name(ino_t netns, unsigned int iface_index);
+
+struct netns {
+	ino_t inode;
+	size_t n_ifaces;
+	struct iface *ifaces;
+};
+
 static int netns_compare(const void *a, const void *b)
 {
-	if (*(ino_t *)a < *(ino_t *)b)
-		return -1;
-	else if (*(ino_t *)a > *(ino_t *)b)
-		return 1;
-	else
-		return 0;
+	const struct netns *netns_a = a;
+	const struct netns *netns_b = b;
+
+	return netns_a->inode - netns_b->inode;
+}
+
+static void netns_free(void *netns)
+{
+	struct netns *nsobj = netns;
+
+	if (nsobj->n_ifaces > 0)
```

Comment:
Is the conditional needed? Freeing NULL is fine

Reply:
Done. I deleted the `n_faces` member. Instead, I introduced `index == 0` sentinel value as you suggested.